### PR TITLE
Update the json to enfore promise rejection with errors and reformat with english

### PR DESCRIPTION
**Pull request by jeffheifetz**

### CODE REVIEW

File: eslintrc.json [hunk 1]
```diff
@@ -2,114 +2,115 @@
...
-    "no-use-before-define": 0,
-    "no-with": 2,
+    "no-extend-native": "error",
+    "no-irregular-whitespace": "error",
+    "no-mixed-spaces-and-tabs": "error",
+    "no-shadow": "error",
+    "no-trailing-spaces": "error",
+    "no-undef": "error",
+    "no-unused-vars": "error",
+    "no-use-before-define": "off",
```
Comment (jeffheifetz) [inline eslintrc.json]: I'd rather turn this one on personally, especially as we use more ES6 let and const that aren't hoisted.

File: eslintrc.json [hunk 2]
```diff
@@ -2,114 +2,115 @@
...
+    "dot-notation": "error",
+    "eol-last": "error",
+    "eqeqeq": "error",
+    "guard-for-in": "error",
     "indent": [
-      2,
+      "error",
       2,
       {
         "SwitchCase": 1
```
Comment (jeffheifetz) [inline eslintrc.json]: I'd like to also enable VariableDeclarator, ArrayExpression and ObjectExpression added here.

The current indent block is actually duplicated from [standard](https://github.com/feross/eslint-config-standard/blob/master/eslintrc.json#L52) 

see http://eslint.org/docs/rules/indent

Comment (AerialX) [reply eslintrc.json]: As we discussed, I agree on using ArrayExpression and ObjectExpression rules to enforce consistency in multi-line expressions.

File: eslintrc.json (hunk 1 above)
Comment (AerialX) [reply eslintrc.json]: I'm fine with turning `no-use-before-define` on.

**Pull request merged**